src/Components/Rides/RiderSignup/RiderSignup.tsx

```diff
@@ -2,36 +2,71 @@
  * Rider signup form.
  */
 
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import { Button, Col, Container, Form, Row } from 'react-bootstrap';
 import { Turnarounds } from '../../../Constants/RidesConstants';
+import { useNavigate } from 'react-router-dom';
 
 import './RiderSignup.scss';
-import 'css/common/forms.scss'
+import 'css/common/forms.scss';
 
 export const RiderSignup = () => {
-  const [disableApartmentTurnaroundTextInput, setDisableApartmentTurnaroundTextInput] = useState(true);
+  const navigate = useNavigate();
+
+  const [
+    disableApartmentTurnaroundTextInput,
+    setDisableApartmentTurnaroundTextInput,
+  ] = useState(true);
+  const [formSubmitted, setFormSubmitted] = useState(false);
+
+  const handleSubmit = () => {
+    // if validated then submit form
+    setFormSubmitted(true);
+  };
+
+  useEffect(() => {
+    let timer = setTimeout(() => {
+      if (formSubmitted) {
+        navigate('/');
+      }
+    }, 4000);
+    return () => clearTimeout(timer);
+  }, [formSubmitted, navigate]);
+
+  if (formSubmitted) {
+    // insert thank you page
+    return <div>Thank You!</div>;
+  }
 
   return (
     <Container>
       <Col className={'mx-auto text-center'} lg={'8'}>
         <span className={'signup-form-title'}>Sign up for a ride to Church!</span>
 
-        <Form className={'text-center'}>
+        <Form className={'text-center'} onSubmit={handleSubmit}>

```

Comment:
Form submit stuff isn't really in the scope of this assignment. We'll have a separate task to handle the redirect after users sign up (this is probably after the Thank You page gets finished.

Reply:
ohh got it. i thought i had to do everything on the trellis task